File: backend/src/main/java/de/bucheeinfach/backend/models/Course.java
package de.bucheeinfach.backend.models;

import de.bucheeinfach.backend.models.enums.CourseStatus;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Document(collection = "courses")
@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Course {
    @Id
    private String id;
    @DBRef
    private Program program;
    @DBRef
    private Location location;
    private int quota;
    private String trainer;
    private double price;
    private LocalDate startDate;
    private LocalDate endDate;
    private String duration;
    private String timePlan;
    private String timePlanExcepted;
    private CourseStatus status;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: frontend/src/types/Course.ts
import {Program} from "./Program.ts";
import {Location} from "./Location.ts";

export type Course = {
    id: string,
    programId: string,
    programName: string,
    locationId: string,
    locationName: string,
    programDescription: string,
    programImageUrl: string,
    quota: number,
    trainer: string,
    startDate: Date,
    endDate: Date,
    price: number,
    duration: string,
    timePlan: string,
    timePlanExcepted: string,
    status: string,
    createdAt: Date,
    updatedAt: Date
    program: Program,
    location: Location
}